cpp/src/cuda/cuda_matrix_vector_multiply.cu

```diff
@@ -64,9 +66,9 @@ namespace Nice {
       CUDA_CALL(cudaMemset(d_y, 0, m * sizeof(T)));
 
       // Launch kernel here
-      //dim3 dimBlock(BLOCK_SIZE *BLOCK_SIZE);
-      //dim3 dimGrid((a.rows() / dimBlock.x) * (a.cols() / dimBlock.y));
-      CudaMVKernel<<<m, 256>>>(d_a, d_x, d_y, m, k);
+      dim3 dimBlock(BLOCK_SIZE);
+      dim3 dimGrid(std::ceil((float)m / (BLOCK_SIZE)));
```

Comment:
A conventional way of set up the grid size without using ceiling function would be:

(m + BLOCK_SIZE -1) / BLOCK_SIZE

Try some examples of m and see if this gives you the same result as using std::ceil()